### PR TITLE
Session resumption

### LINKED ISSUE #184: TLS 1.3 - Session resumption
support for PSK and NewSessionTicket based session resumption in TLS 1.3 and test script that performs it

blocked by #183, #190

### PR DESCRIPTION
**Pull request by tomato42**
### Description

Implement support for session resumption in TLS 1.3

### Motivation and Context

fixes #184 

### Checklist

- [X] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [X] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [X] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [X] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [X] OpenSSL
  - [x] NSS
  - [x] GnuTLS
- [X] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/427)

### CODE REVIEW

File: scripts/test-tls13-session-resumption.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+from __future__ import print_function
+import traceback
+import sys
+import getopt
+from itertools import chain, islice
+
+from tlsfuzzer.runner import Runner
+from tlsfuzzer.messages import Connect, ClientHelloGenerator, \
+        ClientKeyExchangeGenerator, ChangeCipherSpecGenerator, \
```
Comment (Jakuje) [inline scripts/test-tls13-session-resumption.py]: If I see right, the `ChangeCipherSpecGenerator` is unused in this file

File: scripts/test-tls13-session-resumption.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        ExpectServerHelloDone, ExpectChangeCipherSpec, ExpectFinished, \
+        ExpectAlert, ExpectApplicationData, ExpectClose, \
+        ExpectEncryptedExtensions, ExpectCertificateVerify, \
+        ExpectNewSessionTicket, srv_ext_handler_supp_vers, \
+        gen_srv_ext_handler_psk, srv_ext_handler_key_share
+
+from tlslite.constants import CipherSuite, AlertLevel, AlertDescription, \
+        TLS_1_3_DRAFT, GroupName, ExtensionType, SignatureScheme, \
+        PskKeyExchangeMode
+from tlslite.keyexchange import ECDHKeyExchange
```
Comment (Jakuje) [inline scripts/test-tls13-session-resumption.py]: `ECDHKeyExchange` is not use din the code either.

File: scripts/test-tls13-session-resumption.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+from tlsfuzzer.utils.lists import natural_sort_keys
+from tlsfuzzer.utils.ordered_dict import OrderedDict
+from tlslite.extensions import KeyShareEntry, ClientKeyShareExtension, \
```
Comment (Jakuje) [inline scripts/test-tls13-session-resumption.py]: Unused `KeyShareEntry`.

**Comment by tomato42** [comment]
both failures were unrelated; `scripts/test-tls13-finished.py` with the large pad is known to be problematic (#431 among others), rescheduled

**Comment by tomato42** [comment]
Thanks for the review!

**Pull request merged**